internals/overlord/planstate/manager.go

```diff
@@ -181,6 +181,7 @@ func (m *PlanManager) updatePlanLayers(layers []*plan.Layer) (*plan.Plan, error)
 		Services:   combined.Services,
 		Checks:     combined.Checks,
 		LogTargets: combined.LogTargets,
+		Sections:   combined.Sections,
```

Comment:
I don't quite understand this PR. Are we really adding hundreds of lines of tests just to test this 1 line of code, that `Sections` is wired up correctly? If so, I'd argue it's not worth it.

Reply:
Hi Ben, just to make sure I understand your concern: The ```planstate``` manager builds on top of the ```plan``` package to implement the ```CombineLayers``` and ```AppendLayers``` functions (simple, but there are some additional logic added, which feels not ideal to not cover in a test). This is the reason the current planstate testsuite covers them, but only for built-in sections.

This change you commented on here was actually a result of the added tests failing (I discovered that only afterwards).

Instead of having separate ```Load```, ```CombineLayers``` and ```AppendLayers``` tests, I have now merged it with the built-in tests, which makes the delta much smaller. There is still however the need for a test extension to be define, which I kept as it.

Alternatively, I could simply include this line in the ```plan``` PR, and drop adding extension coverage at this level ?

Reply:
Ah, I see now, thanks.